ion-schema/src/constraint.rs

```diff
@@ -130,7 +133,7 @@ impl Constraint {
             .map(|id| {
                 VariablyOccurringTypeRef::new(
                     TypeReference::new(*id, NullabilityModifier::Nothing),
-                    Range::required(),
+                    UsizeRange::new_single_value(1),
```

Comment:
🗺️ The concepts of "required" and "optional" don't inherently belong to ranges, so we're using `new_single_value(1)` here. There's also a `UsizeRange::zero_or_one()` that replaces `Range::optional()`.

Reply:
Instead of `new_single_value(1)` `one()` might make sense here and it also aligns with `zero_or_one()`

Reply:
The benefit of `zero_or_one()` (and the only reason I created it) is that it returns a `UsizeRange` whereas `new_inclusive(0, 1)` returns a `IonSchemaResult<UsizeRange>`. Both `new_single_value(1)` and `one()` would return a `UsizeRange`, so the same motivation isn't there.

I'd like to defer a `one()` function for now (if you don't mind), but I could be convinced otherwise.

Reply:
I think having `new_single_value()` is fine but when we have something like `zero_or_one` user might expect a similar `one` which helps create required occurs. But this isn't a huge issue as we atleast have an alternative API `new_single_value().